cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -724,6 +732,17 @@ private V1ObjectMeta createJob(V1ObjectMeta metadata, Map<String, RuntimeSpecifi
       if (e.getCode() == HttpURLConnection.HTTP_CONFLICT) {
         LOG.warn("The kubernetes job already exists : {}. Ignoring resubmission of the job.", e.getResponseBody());
       } else {
+        // Clean up config maps
+        LOG.info("Deleting configmaps for runId {} in namespace {}", twillRunId.getId(), programRuntimeNamespace);
+        try {
+        coreV1Api.deleteCollectionNamespacedConfigMap(programRuntimeNamespace, null, null, null, null, null,
+                                                      KubeTwillRunnerService.RUN_ID_LABEL + "=" + twillRunId.getId(),
+                                                      null, null, null, null, null, null, null);
+        } catch (ApiException ex) {
+          LOG.warn("Failed to delete configmaps for runId {} in namespace {}",
```

Comment:
if this delete fails for whatever reason, we need some sort of janitor that cleans them up. There is already a KubeJobCleaner class for a similar purpose.